File: 스택+큐/프린터.py
def solution(priorities, location):
    answer = 0
    index = []
    highest = priorities[0]
    while(len(priorities) > 0):
        if priorities[0] < max(priorities):
            priorities.append(priorities.pop(0))
            if location == 0:
                location = len(priorities) - 1
            else:
                location -= 1
        else:
            priorities.pop(0)
            answer += 1
            if location == 0:
                return answer
            location -= 1

File: src/stackQueue/프린터.java
package stackQueue;

import java.util.*;

public class 프린터 {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());

        Arrays.stream(priorities).forEach(p->pq.offer(p));

        while(!pq.isEmpty()){
            for(int i = 0 ; i < priorities.length ; i++){
                if(pq.peek() == priorities[i]){
                    pq.poll();
                    answer++;
                    if(location == i)
                        return answer;
                }
            }

        }

        return answer;

    }
}
